### DISCUSSION TITLE
I have some problems for phoebe

### DISCUSSION THREAD

**Discussion by AstroJLin** [phoebe-project/phoebe2#505] [Questions]
Hi Kyle,
1、I want to know that the difference between phoebe and ELC (Orosz & Hauschildt 2000). 
2、I wonder whether the phoebe is able to model both ellipsoidal variations and irradiation in redback millisecond pulsar binaries system. If the phoebe can do, I should how to use phoebe to model the light curve.
3、I try to interpret the light curves as ellipsoidal variability in the Black Hole Binary. My code as follow:
b = phoebe.default_binary()
b.add_dataset('lc', times=times, fluxes=fluxes, sigmas=sigmas_lc)
b.set_value('distortion_method', component='secondary', value='none')
b.set_value('pblum', component='primary', value=0.5*np.pi)
b.set_value('teff@secondary', 4000)
b.run_compute(model='distortion_method_none')
_ = b.plot(x='phases', model='distortion_method_none', show=True).
 However, the fluxes, and flux variations are obviously too high compare to the blue curve.
![Figure_3](https://user-images.githubusercontent.com/86318178/126893510-17e90d02-9186-4e2b-86f4-f8a9c363017b.png)

**Comment by kecnry** [comment]
> 1、I want to know that the difference between phoebe and ELC (Orosz & Hauschildt 2000).

See the [compute comparison table](http://phoebe-project.org/docs/latest/examples/compute_comparison_table) and the discussion in the [2.3 release paper](http://phoebe-project.org/publications/2020Conroy+) for an overview on the major differences. 

> 2、I wonder whether the phoebe is able to model both ellipsoidal variations and irradiation in redback millisecond pulsar binaries system. If the phoebe can do, I should how to use phoebe to model the light curve.

phoebe will not be able to handle the pulsar itself, but can handle both ellipsoidal variations and irradiation (although may need to fallback on blackbody atmosphere approximations).

> 3、I try to interpret the light curves as ellipsoidal variability in the Black Hole Binary. My code as follow:
> b = phoebe.default_binary()
> b.add_dataset('lc', times=times, fluxes=fluxes, sigmas=sigmas_lc)
> b.set_value('distortion_method', component='secondary', value='none')
> b.set_value('pblum', component='primary', value=0.5*np.pi)
> b.set_value('teff@secondary', 4000)
> b.run_compute(model='distortion_method_none')
> _ = b.plot(x='phases', model='distortion_method_none', show=True).
> However, the fluxes, and flux variations are obviously too high compare to the blue curve.

You can try manually adjusting or fitting other parameters which affect the ellipsoidal amplitude (especially the radius of the distorted star), but I'm not sure you'll be able to reproduce that amplitude or signal with ellipsoidal variations alone.

Hope this helps!
-Kyle

**Reply by AstroJLin** [reply]
Hi Kyle,

Thank you very much for your reply. 

I try manually adjusting the radius of the distorted star according to your suggestion. However, when I fix the distorted star to be a small (3e-6 Rsun), cold (Teff=300 K) blackbody, use the simplest and fastest eclipse model (eclipse_method'= only_horizon),  the phoebe show atmosphere parameters out of bounds.
My code as follow:
b = phoebe.default_binary()

b.set_value('distortion_method', component='secondary', value='none')
b.set_value('ecc', 0)
b.set_value('incl@binary', 70)
b.set_value('q', 0.3)
b.set_value('teff@primary', 7000)
b.set_value('sma@binary', 60) 
b.set_value('requiv@primary', 29) 

b['eclipse_method'] = 'only_horizon'
b.set_value('requiv@secondary', 3.e-6)
b.set_value('teff@secondary', 300)

b.add_dataset('lc', times=np.linspace(0,1,101))
b.run_compute(model='distortion_method_none')
_ = b.plot(kind='lc', model='distortion_method_none', show=True)
![Screenshot from 2021-07-31 23-34-14](https://user-images.githubusercontent.com/86318178/127745995-0ec3c628-6d06-404a-9e85-6ba49d8ad1f5.png)

At the same time, when I try to run the Adopt from estimators in Inverse Problem (General Workflow and Examples), I also meet this problem for atmosphere parameters out of bounds.
![Screenshot from 2021-08-01 00-02-39](https://user-images.githubusercontent.com/86318178/127745999-32fac454-cecb-4ad9-bdce-db179da4a1ee.png)

Should I how to solve this problem under that  it does not influences the distortion and dynamics of the other component as little as possible, I hope I can get an answer.

Thanks again

**Reply by kecnry** [reply]
It should work if you change to atm='blackbody' and ld_coeffs='manual' and then provide the limb-darkening coefficients.

**Reply by AstroJLin** [reply]
Hi Kyle,

I change to b['atm@secondary'] = 'blackbody' and b['ld_coeffs@secondary'] = 'manual' and then provide b.set_value_all('ld_coeffs@primary', [0.5, 0.5]), it does not work yet.
![Screenshot from 2021-08-07 15-33-59](https://user-images.githubusercontent.com/86318178/128592742-de012932-d7ee-464b-8b61-f21d606cd7ee.png)


However, It can work when I change b['atm@secondary'] = 'blackbody', b['ld_mode@secondary'] = 'manual' and b['ld_mode_bol@secondary'] = 'manual'.

I don't know what caused this problem, could you help me explain this problem?

Thanks again

**Reply by kecnry** [reply]
Yes, you may have to change `ld_mode_bol` as well (ld_mode_bol is the bolometric limb-darkening and is only used if you have irradiation enabled).  That will expose `ld_coeffs_func` and `ld_coeffs_bol` parameters which you should set to something reasonable.
